### PR TITLE
Add comment about caching config model values.

### PR DESCRIPTION
**Pull request by doctoryes**
@cpennington @nasthagiri @ormsbee  Please review. Does this text capture the conversation that happened this morning?

### CODE REVIEW

File: config_models/models.py [hunk 1]
```diff
@@ -10,10 +10,18 @@
...
+# But, depending on the cache backend configuration, the old value may still exist in some caches.
 try:
+    # Use the Django settings-configured cache, which could be:
+    #  - memcache or another external networked cache (per-cluster)
+    #  - file cache (per-machine)
+    #  - local memory cache (per-process)
+    #  - other
     cache = caches['configuration']  # pylint: disable=invalid-name
 except InvalidCacheBackendError:
+    # If no caches configured, use the default local-memory (per-process) cache.
```
Comment (cpennington) [inline config_models/models.py]: The default cache also just depends on a configuration value (it's not always per-process).

**Comment by cpennington** [comment]
This should probably be in the README/docs for ConfigModel, as well as a comment in the code.

File: config_models/models.py [hunk 2]
```diff
@@ -10,10 +10,18 @@
 
 from rest_framework.utils import model_meta
 
-
+# Config model values are cached. The caching backend configuration used has implications for how the
+# values are changed in an active release. When a new value is set, its cached value is deleted.
+# But, depending on the cache backend configuration, the old value may still exist in some caches.
```
Comment (nasthagiri) [inline config_models/models.py]: The amount of time the old value continues to exist _may_ depend on the value of `ConfigurationModel.cache_timeout`.

File: config_models/models.py [hunk 3]
```diff
@@ -10,10 +10,18 @@
...
 try:
+    # Use the Django settings-configured cache, which could be:
+    #  - memcache or another external networked cache (per-cluster)
+    #  - file cache (per-machine)
+    #  - local memory cache (per-process)
+    #  - other
     cache = caches['configuration']  # pylint: disable=invalid-name
 except InvalidCacheBackendError:
+    # If no caches configured, use the default local-memory (per-process) cache.
     from django.core.cache import cache  # pylint: disable=ungrouped-imports
```
Comment (nasthagiri) [inline config_models/models.py]: The cache used is determined by the `default` cache configured in `CACHES`: https://docs.djangoproject.com/en/1.10/ref/settings/#caches

File: config_models/models.py (hunk 2 above)
Comment (doctoryes) [inline config_models/models.py]: Added this to the text.

File: config_models/models.py (hunk 3 above)
Comment (doctoryes) [inline config_models/models.py]: Added.

File: config_models/models.py (hunk 1 above)
Comment (doctoryes) [inline config_models/models.py]: Clarified.

**Comment by doctoryes** [comment]
@cpennington @nasthagiri All comments addressed - plus I added in the names of the original authors of this code into AUTHORS. (I don't want others to think I wrote this code myself - it was an oversight to only put my name in AUTHORS before.)

**Comment by doctoryes** [comment]
Oh - and I got the author list here:
https://github.com/edx/edx-platform/commits/5cb129e222270463dc6a2bbd726fd7a1ade0e73b/common/djangoapps/config_models

**Pull request merged**